controllers/configuration/backend/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/oam-dev/terraform-controller/controllers/configuration/backend/util"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	tfstateKey = "tfstate"
+	workspace  = "default"
+)
+
+// Client is the client used to communicate with the Terraform kubernetes backend
+// Modified based on Hashicorp code base https://github.com/hashicorp/terraform/blob/1846a4752ee8056affad16c583b3072bc55feebd/internal/backend/remote-state/kubernetes/backend.go#L188
+// Licensed under Mozilla Public License 2.0
+type Client struct {
+	// The fields below are set from configure
+	kubernetesSecretClient dynamic.ResourceInterface
+	namespace              string
+	labels                 map[string]string
+	nameSuffix             string
+}
+
+// Get fetches the state json from the Terraform kubernetes backend
+// Modified based on Hashicorp code base https://github.com/hashicorp/terraform/blob/1846a4752ee8056affad16c583b3072bc55feebd/internal/backend/remote-state/kubernetes/backend.go#L188
+// Licensed under Mozilla Public License 2.0
+func (c *Client) Get(ctx context.Context) ([]byte, error) {
```

Comment:
What do you fetch the state? By default, Terraform itself can read and write the state.

Reply:
We need to get the variables from the state json.